improver_tests/spotdata/test_SpotManipulation.py

```diff
@@ -357,3 +451,111 @@ def test_neighbour_selection_method_setting(kwargs, expected):
+
+    forecast = spot_variable(np.arange(273, 282))
+    forecast.coord("wmo_id").points = ["None", "None"] + [
+        f"{item:05d}" for item in range(2, 9)
+    ]
+
+    result = SpotManipulation(subset_coord="wmo_id")([forecast, neighbour_cube])
+
+    assert_array_equal(result.data, expected)
+    assert_array_equal(result.coord("wmo_id").points, expected_ids)
+
+
+@pytest.mark.parametrize(
+    "neighbour_data", [np.array([[[1, 2, 3], [0, 0, 0], [5, 10, 15]]])],
+)
+def test_spot_subset_coord_not_found(neighbour_cube):
+    """Test exception raised for unknown subsetting coordinate."""
+
+    forecast = spot_variable(np.arange(273, 282))
+    with pytest.raises(ValueError, match="Subset_coord not found in neighbour cube."):
+        SpotManipulation(subset_coord="kittens")([forecast, neighbour_cube])
+
+
+@pytest.mark.parametrize(
+    "neighbour_data", [np.array([[[1, 2, 3], [0, 0, 0], [5, 10, 15]]])],
+)
+def test_no_spots_returned(neighbour_cube):
+    """Test exception raised if subsetting a spot forecast results in no
+    data being returned. This maybe because the subsetting coordinate does not
+    exist on the forecast cube, or because there are no overlapping sites
+    between the forecast and neighbour cube."""
+
+    # No shared ID coordinate
+    forecast = spot_variable(np.arange(273, 282))
+    forecast.coord("wmo_id").rename("kittens")
+    with pytest.raises(ValueError, match="No spot sites retained after subsetting."):
+        SpotManipulation(subset_coord="wmo_id")([forecast, neighbour_cube])
+
+    # No overlapping sites
+    forecast = spot_variable(np.arange(273, 282))
+    forecast.coord("wmo_id").points = [f"{item:05d}" for item in range(101, 110)]
+    with pytest.raises(ValueError, match="No spot sites retained after subsetting."):
+        SpotManipulation(subset_coord="wmo_id")([forecast, neighbour_cube])
+
+
+@pytest.mark.parametrize(
+    "neighbour_data", [np.array([[[1, 2, 3], [0, 0, 0], [5, 10, 15]]])],
+)
+def test_spot_subset_lapse_rate_exception(neighbour_cube):
+    """Test exception raised if attempting to apply a lapse rate to an existing
+    spot forecast. This could be implemented but has not been as we have no use
+    case at present. The tests here cover a fixed lapse rate or a lapse rate
+    provided as a cube."""
+
+    msg = (
+        "Lapse rate adjustment when subsetting an existing spot "
+        "forecast cube has not been implemented."
+    )
+
+    forecast = spot_variable(np.arange(273, 282))
+    with pytest.raises(NotImplementedError, match=msg):
+        SpotManipulation(
+            subset_coord="wmo_id",
+            fixed_lapse_rate=0.001,
+            apply_lapse_rate_correction=True,
+        )([forecast, neighbour_cube])
+
+    lapse_rate_cube = forecast.copy()
+    with pytest.raises(NotImplementedError, match=msg):
+        SpotManipulation(subset_coord="wmo_id", apply_lapse_rate_correction=True)(
+            [forecast, lapse_rate_cube, neighbour_cube]
```

Comment:
Does this test cover it, or are you looking for something different?

https://github.com/metoppv/improver/pull/2002/files?show-viewed-files=true&file-filters%5B%5D=#diff-2279bb0a43aa2bca99e92159002ce150d4b68665601ab9c048409d3d36b01a48R265

Reply:
Also, please remember to check out the acceptance test data:
https://github.com/metoppv/improver_test_data/pull/46

Reply:
> Also, please remember to check out the acceptance test data: [metoppv/improver_test_data#46](https://github.com/metoppv/improver_test_data/pull/46)

This is what I did, but I rebased with master to get all the tests to pass. There has been an addition of a bias correction KGO since you opened the branch. 

Unfortunately the link to the test you're trying to show me doesn't work.

Reply:
> Does this test cover it, or are you looking for something different?
> 
> https://github.com/metoppv/improver/pull/2002/files?show-viewed-files=true&file-filters%5B%5D=#diff-2279bb0a43aa2bca99e92159002ce150d4b68665601ab9c048409d3d36b01a48R265

Yes it does, thanks, I'd missed that.